### PR TITLE
make live default and add live docs (#72, #77)

### PR DESCRIPTION
**Pull request by joehand**
* Live by default
* document `archive.live`

### CODE REVIEW

**Comment by juliangruber** [comment]
it would be cool though if there was a test where the `live:true` can now be removed

also travis fails in cases where `opts` isn't defined

**Comment by juliangruber** [comment]
otherwise +1!

**Comment by karissa** [comment]
great!

File: archive.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function Archive (drive, key, opts) {
 
   this.options = opts || {}
   this.drive = drive
-  this.live = this.options.live = !!this.options.live
+  this.live = this.options.live = opts.hasOwnProperty('live') ? !!this.options.live : true
```
Comment (mafintosh) [inline archive.js]: i'd prefer `this.options.live !== false`

**Comment by joehand** [comment]
> it would be cool though if there was a test where the `live:true` can now be removed

@juliangruber do you mean removing `live:true` from existing test? I could do that for [this one](https://github.com/mafintosh/hyperdrive/blob/master/test/overwrite.js#L9).

**Comment by juliangruber** [comment]
awesome, +1!

File: test/overwrite.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ var concat = require('concat-stream')
 test('overwrite', function (t) {
   t.plan(2)
   var drive = hyperdrive(memdb())
-  var archive = drive.createArchive(undefined, { live: true })
+  var archive = drive.createArchive(undefined)
```
Comment (mafintosh) [inline test/overwrite.js]: remove undefined as well. was just a spacer for the options map :)

**Pull request merged**